gtsfm/evaluation/metrics.py

```diff
@@ -167,7 +167,7 @@ def _create_summary(self, data: np.ndarray) -> Dict[str, Any]:
         """
         if data.ndim != 1:
             raise ValueError("Metric must be a 1D distribution to get summary.")
-        if data.size == 0:
+        if data.size == 0 or np.isnan(data).all():
             return {"min": np.NaN, "max": np.NaN, "median": np.NaN, "mean": np.NaN, "stddev": np.NaN}
```

Comment:
could you change to something like this? 
```
nan_summary ={"min": np.NaN, "max": np.NaN, "median": np.NaN, "mean": np.NaN, "stddev": np.NaN}
if data.size == 0:
    log warning ("{metric_name} is empty, summary contains NaNs.")
    return nan_summary
if np.isnan(data).all():
   log warning ("all values in {metric_name} are NaNs, hence summary contains NaNs")
   return nan_summary
```
I am a bit undecided about whether we need a NaN summary or just an empty summary, or no summary at all for these cases.